modules/reindex/src/main/java/org/elasticsearch/index/reindex/RestReindexAction.java

```diff
@@ -147,6 +174,31 @@ protected ReindexRequest buildRequest(RestRequest request) throws IOException {
         }
     }
 
+    private static String extractString(Map<String, Object> source, String name) {
+        Object value = source.remove(name);
```

Comment:
There is step missing - I should be failing the request if the map is non-empty because the user sent syntax we don't understand.